lib/zclient.c

```diff
@@ -2593,6 +2739,175 @@ int lm_label_manager_connect(struct zclient *zclient, int async)
 	return -1;
 }
 
+/**
+ * Connect to srv6 manager in a syncronous way
+ *
+ * It first writes the request to zcient output buffer and then
+ * immediately reads the answer from the input buffer.
+ *
+ * @param zclient Zclient used to connect to srv6 manager (zebra)
+ * @result Result of response
+ */
+int srv6_manager_connect(struct zclient *zclient)
+{
+	struct stream *s;
+	int ret, result = 0;
+	uint16_t cmd = ZEBRA_SRV6_MANAGER_CONNECT;
+
+	if (zclient_debug)
+		zlog_debug("Connecting to SRv6 Manager");
+
+	if (zclient->sock < 0) {
+		zlog_debug("%s: invalid zclient socket", __func__);
+		return -1;
+	}
+
+	/* send request */
+	s = zclient->obuf;
+	stream_reset(s);
+	zclient_create_header(s, cmd, VRF_DEFAULT);
+
+	stream_putc(s, zclient->redist_default); /* proto */
+	stream_putw(s, zclient->instance);       /* instance */
+	stream_putw_at(s, 0, stream_get_endp(s));
+	ret = writen(zclient->sock, s->data, stream_get_endp(s));
+	if (ret < 0) {
+		flog_err(EC_LIB_ZAPI_SOCKET, "Can't write to zclient sock");
+		close(zclient->sock);
+		zclient->sock = -1;
+		return -1;
+	}
+	if (ret == 0) {
+		flog_err(EC_LIB_ZAPI_SOCKET, "Zclient sock closed");
+		close(zclient->sock);
+		zclient->sock = -1;
+		return -1;
+	}
+	if (zclient_debug)
+		zlog_debug("SRv6 Manager connect request sent (%d bytes)", ret);
+
+	/* read response */
+	if (zclient_read_sync_response(zclient, cmd)
+	    != 0)
+		return -1;
+
+	s = zclient->ibuf;
+
+	/* read instance and proto */
```

Comment:
you've just asked for this synchronous exchange - how could the proto and instance be wrong? what is the purpose of this?

Reply:
This line has been eliminated by dropping srv6_manager_connect.